workspaces/config/lib/definitions/index.js

```diff
@@ -71,10 +71,22 @@ const nerfDarts = [
   'username', // Does not have a config
 ]
 
+// Environment variables that can affect npm's behavior but are not npm configs
+// These are shown in `npm config list` to help users understand their environment
+const proxyEnv = [
+  'HTTP_PROXY',
```

Comment:
Technically any casing will be picked up by npm here https://github.com/npm/agent/blob/abcc65439978ba2fa377f3e76a9853fb08e62f88/lib/proxy.js#L16

To prevent big surprises in the outside chance someone has set `process.env.Http_proxy` this should be handled in the config list in a similar way to how it's parsed in the agent (and other libs).